### PR TITLE
fix(kumactl): get APIVersions from k8s server

### PR DESCRIPTION
**Pull request by michaelbeaumont**
This wasn't being set correctly when using actual clusters so `.Capabilities.APIVersions` in templates wasn't working correctly with `kumactl`. Which meant that the `GatewayClass` in the templates wasn't ever created, even if Gateway API CRDs are in the cluster.

This code is more or less directly [inlined from helm](https://github.com/helm/helm/blob/5cb9af4b1b271d11d7a97a71df3ac337dd94ad37/pkg/action/action.go#L239) because these aren't public functions.

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: app/kumactl/cmd/install/install_control_plane.go [hunk 1]
```diff
@@ -45,6 +47,81 @@ func (cv *componentVersion) Type() string {
 	return "string"
 }
 
+// getVersionSet retrieves a set of available k8s API versions
```
Comment (slonka) [inline app/kumactl/cmd/install/install_control_plane.go:50]: I'd link to [the code we took this from](https://github.com/helm/helm/blob/5cb9af4b1b271d11d7a97a71df3ac337dd94ad37/pkg/action/action.go#L310), people might not read the PR, assume it's our code and if a bug appears then a person will check the upstream and not try to fix it themselves

File: app/kumactl/cmd/install/install_control_plane_test.go [hunk 2]
```diff
@@ -139,16 +141,13 @@ var _ = Describe("kumactl install control-plane", func() {
 			ExpectMatchesGoldenFiles(actual, filepath.Join("testdata", given.goldenFile))
 		},
 		Entry("should generate Kubernetes resources with default settings", testCase{
-			extraArgs: []string{
-				"--without-kubernetes-connection",
```
Comment (slonka) [inline app/kumactl/cmd/install/install_control_plane_test.go:143]: why are we changing this?

File: test/e2e/gateway/gatewayapi/conformance_test.go [hunk 3]
```diff
@@ -23,15 +23,6 @@ var clusterName = Kuma1
 var minNodePort = 30080
 var maxNodePort = 30089
 
-const gatewayClass = `
```
Comment (slonka) [inline test/e2e/gateway/gatewayapi/conformance_test.go:26]: this seems unrelated, the PR is only about `.Capabilities.APIVersions` not being right

**Review by slonka — COMMENTED** [review]
Left a couple of questions.

File: test/e2e/gateway/gatewayapi/conformance_test.go (hunk 3 above)
Comment (michaelbeaumont) [reply test/e2e/gateway/gatewayapi/conformance_test.go:26]: the problem was that the GatewayClass wasn't being created at install time, which is fixed now. so we no longer need to create this "default kuma Gateway" GatewayClass in e2e tests, it just exists

File: app/kumactl/cmd/install/install_control_plane_test.go (hunk 2 above)
Comment (michaelbeaumont) [reply app/kumactl/cmd/install/install_control_plane_test.go:143]: I'm now applying it to _all_ the Entries. These tests run without an actual Kubernetes API server so I think it only makes sense to have this flag everywhere here

Comment (michaelbeaumont) [reply app/kumactl/cmd/install/install_control_plane_test.go:143]: Of course the actual motivator here is that with this PR we know attempt to query the API server if this flag isn't present. so some of these tests that didn't have the flag before now fail

**Review by slonka — APPROVED** [review]
